src/H5T.c

```diff
@@ -746,6 +770,47 @@ H5T__init_inf(void)
         } /* end for */
     }     /* end if */
 
+    /* Get the long double datatype */
```

Comment:
+/- Inf for `long double` was previously never set, presumably due to IBM `long double` issues and also since there's currently no larger standard type than `long double`, so the values weren't needed for overflow/underflow checks during conversion. However, complex numbers have some special logic that needed +/- Inf for `long double _Complex` conversions, so I set +/- Inf for `long double` and use it to construct +/- Inf for `long double _Complex`. The value will likely be wrong on machines that use the IBM `long double` format until fixes are made to the library, but this shouldn't be any more problematic than the current situation.